### PR TITLE
Empty state message for project overview

### PR DESCRIPTION
**Pull request by spadgett**
Remove some of the empty whitespace and provide help to get the user
started creating an application.

openshift_management_console

### CODE REVIEW

**Comment by spadgett** [comment]
@liggitt please review

**Comment by smarterclayton** [comment]
Like

File: assets/app/styles/_core.less [hunk 1]
```diff
@@ -214,6 +214,21 @@ html, body {
       }
     }
   }
+  .empty-project-msg {
```
Comment (liggitt) [inline assets/app/styles/_core.less]: maybe a less specific class name? we might want to reuse this for other empty state boxes?

Comment (liggitt) [inline assets/app/styles/_core.less]: is this style from the patternfly empty-state pattern?

File: assets/app/views/project.html [hunk 2]
```diff
@@ -7,11 +7,20 @@
       </div>
       <!-- TODO handle things that don't live under services -->
       <alerts alerts="alerts"></alerts>
-      <div ng-if="(services | hashSize) == 0">
-        <div>
-          <em>{{emptyMessage}}</em>
-        </div>
+      <div ng-if="loadingServices">Loading...</div>
```
Comment (liggitt) [inline assets/app/views/project.html]: we don't want to show the "Your project is empty" message if they filtered down project contents to nothing.
Keep this structure:
```
<div ng-if="(services | hashSize) == 0">
  <div ng-if="showGetStarted">
  </div>
  <div ng-if="!showGetStarted">
    <em>{{emptyMessage}}</em>
  </div>
</div>
```

File: assets/app/scripts/controllers/overview.js [hunk 3]
```diff
@@ -40,9 +40,9 @@ angular.module('openshiftConsole')
 
     $scope.labelSuggestions = {};
     $scope.alerts = $scope.alerts || {};
-    $scope.emptyMessage = "Loading...";
+    $scope.loadingServices = true;
```
Comment (liggitt) [inline assets/app/scripts/controllers/overview.js]: keep `emptyMessage` as is

File: assets/app/scripts/controllers/overview.js [hunk 4]
```diff
@@ -55,6 +55,9 @@ angular.module('openshiftConsole')
 
     watches.push(DataService.watch("services", $scope, function(services) {
       $scope.unfilteredServices = services.by("metadata.name");
+      $scope.renderOptions.showSidebarRight =
```
Comment (liggitt) [inline assets/app/scripts/controllers/overview.js]: toggle `$scope.showGetStarted` based on the size of `unfilteredServices`

File: assets/app/styles/_core.less (hunk 1 above)
Comment (spadgett) [inline assets/app/styles/_core.less]: @liggit No, I searched, but couldn't find anything about paternally empty-states.

File: assets/app/views/project.html (hunk 2 above)
Comment (spadgett) [inline assets/app/views/project.html]: @liggitt There's a different message when all projects are hidden by filters, see updateFilterWarning(). But you're right that I'm incorrectly showing get started as well. Good catch. I'll fix.

**Comment by spadgett** [comment]
@liggitt Thanks for the feedback. Have another look.

File: assets/app/scripts/controllers/overview.js [hunk 5]
```diff
@@ -55,6 +55,10 @@ angular.module('openshiftConsole')
 
     watches.push(DataService.watch("services", $scope, function(services) {
       $scope.unfilteredServices = services.by("metadata.name");
+      var isEmpty = Object.keys($scope.unfilteredServices).length === 0;
+      $scope.renderOptions.showSidebarRight = !isEmpty;
+      $scope.showGetSarted = isEmpty;
```
Comment (liggitt) [inline assets/app/scripts/controllers/overview.js]: Typo

**Comment by liggitt** [comment]
s/You'll need to/You can/

**Comment by spadgett** [comment]
@liggitt thanks, I made the updates.

File: assets/app/scripts/controllers/overview.js [hunk 6]
```diff
@@ -55,6 +55,10 @@ angular.module('openshiftConsole')
 
     watches.push(DataService.watch("services", $scope, function(services) {
       $scope.unfilteredServices = services.by("metadata.name");
+      var isEmpty = Object.keys($scope.unfilteredServices).length === 0;
+      $scope.renderOptions.showSidebarRight = !isEmpty;
+      $scope.showGetStarted = isEmpty;
```
Comment (liggitt) [inline assets/app/scripts/controllers/overview.js]: sorry to make you tweak it one more time... group this option under `renderOptions`?

File: assets/app/styles/_core.less [hunk 7]
```diff
@@ -138,6 +138,22 @@ html, body {
...
+  background-color: #ffffff;
+  border-radius: @border-radius-sm;
+  box-shadow: @osc-object-box-shadow;
+  font-weight: 300;
+  margin: auto;
+  max-width: 600px;
+  padding: 30px;
+  text-align: center;
+  p {
+    font-size: @font-size-large;
```
Comment (liggitt) [inline assets/app/styles/_core.less]: does this font size and padding work all the way down to mobile resolutions?

Comment (spadgett) [inline assets/app/styles/_core.less]: @liggitt Looks OK, except the navbar takes a ton of space.

openshift_management_console

**Comment by spadgett** [comment]
@liggitt updated to use $scope.renderOptions.showGetStarted

**Comment by liggitt** [comment]
[merge]

**Pull request merged**